### ISSUE TITLE
Production Android crash

### ISSUE THREAD

**Issue by briblanch** [flutter/flutter#11387]
## Issue
I have recently released an application to production for both iOS and Android. I am currently experiencing a crash in production only on Android and the stack trace that the play store is providing is the following:
```
  native: pc 00000000000540cc  /data/app/com.blanchard.golfiowe-1/lib/arm/libflutter.so
  native: pc 00000000006067bb  /data/app/com.blanchard.golfiowe-1/oat/arm/base.odex
```

Also, the crash seems to be happening only on Samsung Galaxy devices and a DROID Turbo 2 on Android 7.0. My user base is not very large yet so I am not quite sure how many other devices are affected. I have tested on a physical Pixel device and I do not see the crash.

I have recently fixed a crash that I was experiencing on all platforms where there is a TabBarView with an input as a child that has a focus node attached to it and when the active tab moved far enough away from the tab that had the focused input the app crashed (I am guessing because the the input that was still focused was removed from the widget tree). This crash I am experiencing now might be related to that but I can't say for sure.

Any help you can lend is greatly appreciated and thanks so much for building something as awesome as Flutter!

## Flutter Doctor
[✓] Flutter (on Mac OS X 10.12.5 16F2073, locale en-US, channel master)
    • Flutter at /Users/brianblanchard/workspaces/sdks/flutter
    • Framework revision 5756d6293 2017-07-22 06:12:42 -0700
    • Engine revision 5fcfb995bb
    • Tools Dart version 1.25.0-dev.7.0

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.0)
    • Android SDK at /Users/brianblanchard/Library/Android/sdk
    • Platform android-26, build-tools 26.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] iOS toolchain - develop for iOS devices (Xcode 8.3.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.3.3, Build version 8E3004b
    • ios-deploy 1.9.1
    • CocoaPods version 1.2.1

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

**Comment by eseidelGoogle** [comment]
@jason-simmons @a-siva could this be related to the crash we were seeing with Posse's app?

@briblanch to confirm: are you sure that the flutter version mentioned in the doctor output is the same the app was released with?  That flutter version is from 7/22, which is only 2 days ago. :)

**Comment by eseidelGoogle** [comment]
Assuming the Flutter version is correct https://github.com/flutter/engine/wiki/Symbolicating-production-crash-stacks should work to tell us the symbol it crashed on.  Given how short the stack is, I doubt we'll get very useful information from the stack however. :/

**Comment by briblanch** [comment]
@eseidelGoogle I was looking at the output of `git reflog` and I thought I only update once after my last release but after your clarification I am 99% sure it was `489150693`.  Sorry for the confusion, I was in a hurry to get the app out the door and didn't pin the flutter version 😄.

**Comment by eseidelGoogle** [comment]
OK, so https://github.com/flutter/flutter/commits/489150693 means flutter/engine@ 8ad1576f2711b069341756c014924d3aedbb47a8

**Comment by eseidelGoogle** [comment]
```
~/Library/Android/sdk/ndk-bundle/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/bin/arm-linux-androideabi-addr2line -e ~/Downloads/libflutter.so 00000000000540cc                                                                                         00000000000540cc
/b/build/slave/Linux_Engine/build/src/out/android_release/../../flutter/fml/mapping.cc:58
```
Which I'm not sure if that makes any sense or not.

**Comment by eseidelGoogle** [comment]
https://github.com/flutter/engine/blob/master/fml/mapping.cc#L58 doesn't look like a likely line to crash at. :/

**Comment by eseidelGoogle** [comment]
It looks like this is your app: https://play.google.com/store/apps/details?id=com.blanchard.golfiowe we have various Galaxy devices around, we can try and see if one crashes.

**Comment by briblanch** [comment]
@eseidelGoogle that would be great. I will need to set you up with some account info. Would it be alright if I emailed that information to you?

**Comment by eseidelGoogle** [comment]
I think that https://github.com/flutter/engine/pull/3927 might fix this.

@briblanch are you using Firebase in your application or any other Flutter plugins?

**Comment by briblanch** [comment]
@eseidelGoogle I am using Firebase. I will release a new version of my app built with this fix and let you know if the crash is mitigated.

**Comment by briblanch** [comment]
@eseidelGoogle I believe the fix in flutter/engine#3927 has fixed my production crashes. If I see the issue arise again I will create a new issue. Thanks again for the help and you can close this issue if you would like.
